### PR TITLE
Fix issue where captures don't start after restarting from desktop mode

### LINKED ISSUE #8: Replay options grayed out
I don't know if I'm the only one with this issue, but every time I wanna record the last 30 seconds or 5 minutes of game play, the options are all grayed out. While other times it works. 90% of the time it doesn't work, any idea why?

### PR DESCRIPTION
**Pull request by Ratismal**
This PR does the following:
- on psutil functions, try/except so that zombie processes don't crash the function
    - Right now the except clause just passes, I'm not sure what a more elegant way to handle zombie processes is
- On watchdog, capturing is killed when exiting gamemode. We also need to start capturing when re-entering gamemode

Fixes #8

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -46,15 +46,23 @@
 def find_gst_processes():
     pids = []
     for child in psutil.process_iter():
-        if "Decky-Recorder" in " ".join(child.cmdline()):
-            pids.append(child.pid)
+        try:
+            if "Decky-Recorder" in " ".join(child.cmdline()):
+                pids.append(child.pid)
+        except:
```
Comment (safijari) [inline main.py:52]: Please do not use bare except. At a minimum please do `except Exception`. Better way is to except on the specific error you expect to see. Best ways is to figure out what's causing the exception and handling that in the code. Is `child` None some times maybe? or child.cmdline is not returning a list and therefore the join fails?

File: main.py [hunk 2]
```diff
@@ -46,15 +46,23 @@
...
+            # Handle zombie processes?
+            pass
     return pids
 
 
 def in_gamemode():
     for child in psutil.process_iter():
-        if "gamescope-session" in " ".join(child.cmdline()):
```
Comment (safijari) [inline main.py:56]: Same comment as above

File: main.py [hunk 3]
```diff
@@ -274,7 +285,7 @@ async def get_local_fileformat(self):
         return self._fileformat
 
     async def loadConfig(self):
-        logger.info("Loading settings from: {}".format(os.path.join(settingsDir, "settings.json")))
+        logger.info("Loading settings from: {}".format(os.path.join(settingsDir, "decky-loader-settings.json")))
```
Comment (safijari) [inline main.py:288]: I don't believe this is necessary since the settings are already inside a decky-recorder folder.

**Review by safijari — CHANGES_REQUESTED** [review]
Thank you so much for contributing these changes. I'm looking forward to merging this but I think a few items can be improved. I have highlighted those in my comments. Please let me know if you have any questions.

File: main.py (hunk 3 above)
Comment (Ratismal) [reply main.py:288]: Being clear, the settings file is already `decky-loader-settings.json`. All this is doing is making the console log match what the actual file is.

File: main.py (hunk 1 above)
Comment (Ratismal) [reply main.py:52]: It's been a while so I don't remember off the top of my head what the exact error is, but it's to do with zombie processes. I'll have to debug this again to know for sure.

Comment (Ratismal) [reply main.py:52]: Looks like I have some easily accessible logs currently on my steamdeck!
```
2023-11-07 21:07:42,527 - INFO - Traceback (most recent call last):
  File "/home/deck/homebrew/plugins/decky-recorder/main.py", line 118, in start_capturing
    await Plugin.clear_rogue_gst_processes(self)
  File "/home/deck/homebrew/plugins/decky-recorder/main.py", line 80, in clear_rogue_gst_processes
    gst_pids = find_gst_processes()
               ^^^^^^^^^^^^^^^^^^^^
  File "/home/deck/homebrew/plugins/decky-recorder/main.py", line 49, in find_gst_processes
    if "Decky-Recorder" in " ".join(child.cmdline()):
                                    ^^^^^^^^^^^^^^^
  File "/home/deck/homebrew/plugins/decky-recorder/backend/out/psutil/psutil/__init__.py", line 702, in cmdline
    return self._proc.cmdline()
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/deck/homebrew/plugins/decky-recorder/backend/out/psutil/psutil/_pslinux.py", line 1650, in wrapper
    return fun(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/deck/homebrew/plugins/decky-recorder/backend/out/psutil/psutil/_pslinux.py", line 1788, in cmdline
    self._raise_if_zombie()
  File "/home/deck/homebrew/plugins/decky-recorder/backend/out/psutil/psutil/_pslinux.py", line 1693, in _raise_if_zombie
    raise ZombieProcess(self.pid, self._name, self._ppid)
psutil.ZombieProcess: PID still exists but it's a zombie (pid=1275)
```

According to [psutils docs](https://psutil.readthedocs.io/en/latest/#psutil.NoSuchProcess), psutil.ZombieProcess extends from psutil.NoSuchProcess, so only NoSuchProcess needs to be caught.
> Raised by [Process](https://psutil.readthedocs.io/en/latest/#psutil.Process) class methods when no process with the given pid is found in the current process list, or when a process no longer exists. name is the name the process had before disappearing and gets set only if [Process.name()](https://psutil.readthedocs.io/en/latest/#psutil.Process.name) was previously called.
Judging from the description, I don't think any action needs to be taken and the loop can safely be continued. I'll update the code accordingly.

**Review by safijari — APPROVED** [review]

**Comment by safijari** [comment]
I'm not sure how this works. Are you able to merge now that I've approved? If yes, please do so at your convenience. Otherwise let me know and I can merge.

Thank you for your pr

**Comment by Ratismal** [comment]
For anti-abuse reasons, only configured collaborators on a repository are able to do things such merge PRs, so merging it is something you'll have to do. I recommend doing a squash commit so that only one commit is made - the commit links back to the original PR for context, and you're left with a cleaner git commit history.

**Pull request merged**